engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java

```diff
@@ -95,6 +95,26 @@ public String toDisplayString(
     }
   }
 
+  @ExportMessage
+  Type getMetaObject(@CachedLibrary("this") InteropLibrary thisLib) {
+    return EnsoContext.get(thisLib).getBuiltins().dataflowError();
+  }
+
+  @ExportMessage
+  boolean hasMetaObject() {
+    return true;
+  }
+
+  @ExportMessage
+  boolean isException() {
+    return true;
+  }
+
+  @ExportMessage
+  RuntimeException throwException() throws UnsupportedMessageException {
+    return this;
+  }
```

Comment:
Good question. It extends [AbstractTruffleException](https://github.com/enso-org/enso/blob/09c43d36d0e4bdfee00759a83eaad576dfcfd27f/engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java#L24), so it can be an exception. The `throwException` doesn't throw, but really returns it - e.g. it behaves as expected in Enso.

Moreover, when talking to Enso via `org.graalvm.polyglot`, it seems useful to be able to be able to check `Value.isException()` when one evaluates something.

I'd leave `DataflowError` as exception until we find it violates something.